documentation/specs/proposed/BuildCheck-feature-threat-model.md

```diff
@@ -0,0 +1,122 @@
+---
+title: Threat model of MSBuild BuildCheck feature
+---
+
+# BuildCheck Feature Description
+
+The infrastructure within MSBuild allowing pluggability and execution of
+Analyzers and their Rules previously known as "warning waves" and
+"MSBuild Analyzers".
+
+The feature is meant to help customers to improve and understand quality
+of their MSBuild scripts via rules violations reporting. It will allow
+MSBuild to gradually roll out additional rules, as users will be capable
+to configure their opt-in and severity of reports – preventing unwanted
+build breakages. And to equip powerusers to roll out their own quality
+checks – whether for general community or internal enterprise usage.
+
+[Design
+Spec](https://github.com/dotnet/msbuild/blob/main/documentation/specs/proposed/BuildCheck.md)  
+[Architecture](https://github.com/dotnet/msbuild/blob/main/documentation/specs/proposed/BuildCheck-Architecture.md)
+
+# Threats Identification
+
+This feature does not alter existing nor create any new trust
+boundaries.
+
+It is assumed to rely on only trusted sources, be managed by trusted
+operators, and operated on trusted machines.
+
+For this document, we do not address any threats that result from
+violating these conditions.
+
+Acquisition
+
+#### Threat: Supply chain attack on custom analyzer
+
+Custom BuildCheck analyzers are executed during build. If bad external
+actors inject malicious code into it by supply chain attack or somehow
+else, such code can run on build machine, mostly build agent or develop
+box.  
+  
+Mitigation
+
+Custom analyzers are delivered as regular nuget packages by MSBuild \`\<
+PackageReference /\>\` element. This way custom analyzer packages will
+be included in the generated SBOM and Component Government can detect
```

Comment:
This is very good question. Generating SBOM is very complicated stuff and every organization shall take it very seriously. Such SBOM, either generated, manualy maintained or both, shall include both development packages (such as those with MSBuild tasks or roslyn analyzers) and build artifacts dependency packages. 
Since this is esential, IMO, for modem supplay chain security, in scope of this document I assume it is properly in place.

Reply:
Currently we plan to use nuget client as for any other PackageReference entries. The only difference will be in expecting a 'magic' property function call in packaged props file (that's how msbuild will distinguish the analyzers packages during evaluation).
But there is currently no plan for altered/custom resolution of references